psm_lib.py

```
@@ -16,53 +41,335 @@ def calculate_propensity_score(df, treatment_col, covariate_cols):
+            model.fit(X, y)
+            
+            # Predict Propensity Scores
+            ps = model.predict_proba(X)[:, 1]
+            
+            data['ps_score'] = ps
+            # Logit transformation: ln(p / (1-p)) - better for matching linearity
+            # Clip to avoid inf
+            ps_clipped = np.clip(ps, 1e-6, 1 - 1e-6)
+            data['logit_ps'] = np.log(ps_clipped / (1 - ps_clipped))
+            
+            logger.info(f"Propensity scores calculated: mean={ps.mean():.3f}, std={ps.std():.3f}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)